### PR TITLE
Kafka devservices option to create topic-partitions inside the broker

### PR DESCRIPTION
**Pull request by ozangunalp**
Added an IT for Kafka devservices

### CODE REVIEW

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java [hunk 1]
```diff
@@ -60,4 +61,23 @@
     @ConfigItem(defaultValue = "kafka")
     public String serviceName;
 
+    /**
+     * The topic-partition pairs to create in the Dev Services Kafka broker.
+     * After the broker is started, given topics with partitions are created, skipping already existing topics.
+     * For example, <code>quarkus.kafka.devservices.topic-partitions.test=2</code> will create a topic named
+     * {@code test} with 2 partitions.
+     * <p>
+     * The topic creation will not try to repartition existing topics with different partitions.
```
Comment (cescoffier) [inline extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java:70]:
```suggestion
     * The topic creation will not try to re-partition existing topics with a different number of partitions.
```

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java [hunk 2]
```diff
@@ -60,4 +61,23 @@
...
+    @ConfigItem
+    public Map<String, Integer> topicPartitions;
+
+    /**
+     * Timeout in milliseconds for admin client calls used in topic creation.
+     * <p>
+     * Defaults to 2000 ms.
+     */
+    @ConfigItem(defaultValue = "2000")
+    public Integer topicPartitionTimeout;
```
Comment (cescoffier) [inline extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java:81]: Use Duration instead.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks good, I just made 2 comments. 

A section in the documentation would be great.

**Comment by ozangunalp** [comment]
- Added doc section
- Change timeout to Duration and renamed the property

**Review by cescoffier — APPROVED** [review]

**Pull request merged**